modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/RegionClassLoader.java

```diff
@@ -233,23 +226,27 @@ public final URL findResource(final String name) {
   @Override
   public void dispose() {
     registeredClassLoaders.stream().map(c -> c.unfilteredClassLoader).forEach(classLoader -> {
-      try {
-        classLoader.dispose();
-      } catch (Exception e) {
-        final String message = "Error disposing classloader for '{}'. This can cause a memory leak";
-        if (logger.isDebugEnabled()) {
-          logger.debug(message, getArtifactDescriptor().getName(), e);
-        } else {
-          logger.error(message, getArtifactDescriptor().getName());
-        }
-      }
+      disposeClassLoader(classLoader);
     });
-
     registeredClassLoaders.clear();
+    disposeClassLoader(ownerClassLoader);
 
     super.dispose();
   }
 
+  private void disposeClassLoader(ArtifactClassLoader classLoader) {
+    try {
+      classLoader.dispose();
+    } catch (Exception e) {
+      final String message = "Error disposing classloader for '{}'. This can cause a memory leak";
+      if (logger.isDebugEnabled()) {
+        logger.debug(message, getArtifactDescriptor().getName(), e);
+      } else {
+        logger.error(message, getArtifactDescriptor().getName());
```

Comment:
should somehow indicate which call to this method (which inner classlaoder) failed